src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Application/Commands/PriorLearningDataHandlerTests.cs

```diff
@@ -130,63 +127,88 @@ public async Task Handle_WhenRecognisePriorLearningIsNotTrue_ExceptionIsThrown(b
+                e.ErrorMessage == error).Should().Be(true);
         }
 
-        [Test]
-        public async Task Handle_WhenNoDurationReducedByHoursIsSet_ExceptionIsThrown()
+        [TestCase(null, "You must enter the hours, the hours can't be negative, the hours must be 999 or less")]
+        public async Task Handle_WhenNoDurationReducedByHoursIsSet_ExceptionIsThrown(int? cval,  string error)
         {
             fixture = new PriorLearningDataHandlerTestsFixture();
-            fixture.Command.DurationReducedByHours = null;
+            fixture.Command.DurationReducedByHours = cval;
             await fixture.Handle();
 
             var exception = fixture.Exception as DomainException;
             var domainErrors = exception.DomainErrors.ToList();
 
             domainErrors.Count().Should().BeGreaterThan(0);
-            domainErrors.Any(e => e.PropertyName == "DurationReducedByHours" && e.ErrorMessage == "You must enter the hours, the hours can't be negative, the hours must be 999 or less").Should().Be(true);
+            domainErrors.Any(e => 
+            e.PropertyName == "DurationReducedByHours" && e.ErrorMessage == error).Should()
+                .Be(true);
         }
 
-        [Test]
-        public async Task Handle_WhenNoTrainingTotalHoursIsSet_ExceptionIsThrown()
+        [TestCase(null, "You must enter the hours, the hours can't be negative, the hours must be 9999 or less")]
+        [TestCase(-1, "The hours can't be negative")]
+        [TestCase(99999, "The hours entered must be 9999 or less")]
+        public async Task Handle_WhenNoTrainingTotalHoursIsSet_ExceptionIsThrown(int? cval, string error)
         {
             fixture = new PriorLearningDataHandlerTestsFixture();
-            fixture.Command.TrainingTotalHours = null;
+            fixture.Command.TrainingTotalHours = cval;
             await fixture.Handle();
 
             var exception = fixture.Exception as DomainException;
             var domainErrors = exception.DomainErrors.ToList();
 
             domainErrors.Count().Should().BeGreaterThan(0);
-            domainErrors.Any(e => e.PropertyName == "trainingTotalHours" && e.ErrorMessage == "You must enter the hours, the hours can't be negative, the hours must be 9999 or less").Should().Be(true);
+            domainErrors.Any(e => e.PropertyName == "trainingTotalHours" && e.ErrorMessage == error).Should().Be(true);
+        }
+
+
+        [Test]
+        public async Task Handle_WhenRplData_is_Valid()
+        {
+            fixture = new PriorLearningDataHandlerTestsFixture();
+            fixture.Command.TrainingTotalHours = 100;
+            fixture.Command.DurationReducedByHours = 10;
+            fixture.Command.IsDurationReducedByRpl = true;
+            fixture.Command.DurationReducedBy = 1;
+            fixture.Command.CostBeforeRpl = 1000;
+            fixture.Command.PriceReducedBy = 100;
+
+            await fixture.Handle();
+
+            fixture.VerifyRplDataMatchesCommand();
```

Comment:
This is great and tests we save the record but can we also test this code

 if (isDurationReducedByRpl == false)
            {
                PriorLearning.DurationReducedBy = null;
            }

Reply:
Covered this one now